modules/entity_version_history/tests/src/FunctionalJavascript/HistoryOverviewTest.php

```diff
@@ -0,0 +1,199 @@
+
+use Drupal\entity_version_history_test\EventSubscriber\TestHistoryOverviewAlterEventSubscriber;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\node\Entity\Node;
+
+/**
+ * Tests the history overview table.
+ *
+ * @group entity_version_history
+ */
+class HistoryOverviewTest extends WebDriverTestBase {
+
+  /**
+   * An array of node revisions.
+   *
+   * @var \Drupal\node\NodeInterface[]
+   */
+  protected $nodes;
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_history',
+    'entity_version_history_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+      'access entity version history',
+      'edit any first_bundle content',
+      'administer content types',
+    ]);
+    $this->drupalLogin($this->adminUser);
+
+    // Create a history tab setting for the corresponding entity type
+    // and bundle.
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $history_storage = $entity_type_manager->getStorage('entity_version_history_settings');
+    $history_storage->create([
+      'target_entity_type_id' => 'node',
+      'target_bundle' => 'first_bundle',
+      'target_field' => 'field_entity_version',
+    ])->save();
+
+    // We need to invalidate cache because in our route subscriber and our
+    // entity type alter we are depending on the config we created above.
+    $entity_type_manager->clearCachedDefinitions();
+    $this->container->get('router.builder')->rebuild();
+
+    $node = $entity_type_manager->getStorage('node')->create([
+      'type' => 'first_bundle',
+      'title' => 'My test node',
+    ]);
+    $node->save();
+
+    // Get original node.
+    $nodes = [];
+    $nodes[] = clone $node;
+
+    // Create five revisions.
+    $revision_count = 5;
+    for ($i = 0; $i < $revision_count; $i++) {
+      $version_number = $i + 1;
+      // Change the title for each revision.
+      $node->title = 'My test node ' . $version_number;
+      $node->setNewRevision();
+
+      // Increment the version numbers.
+      $node->field_entity_version->major = $version_number;
+      $node->field_entity_version->minor = $version_number;
+      $node->field_entity_version->patch = $version_number;
+
+      // Edit the 1st and 2nd revision with a different user.
+      if ($i < 2) {
+        $editor = $this->drupalCreateUser();
+        $node->setRevisionUserId($editor->id());
+      }
+      else {
+        $node->setRevisionUserId($this->adminUser->id());
+      }
+
+      $node->save();
+
+      // Get each revision.
+      $node = Node::load($node->id());
```

Comment:
No need to reload. Also why you cloning node objects left and right? Also, you dont have node*s* but one node with multiple revisions. Please update everywhere accordingly.

Reply:
I need cloning because we are working on the same object. Without cloning all revision element will have the latest revision ID loaded.

Reply:
Then don;t do it like that. Just make the revisions and then at the load them all and put them in the revisions array.